components/channel_members_rhs/member_list.tsx

```diff
@@ -2,107 +2,94 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import styled from 'styled-components';
-import {CSSTransition} from 'react-transition-group';
+import AutoSizer from 'react-virtualized-auto-sizer';
+import {FixedSizeList} from 'react-window';
+import InfiniteLoader from 'react-window-infinite-loader';
 
 import {UserProfile} from '@mattermost/types/users';
 import {Channel} from '@mattermost/types/channels';
 
 import Member from './member';
 import {ChannelMember} from './channel_members_rhs';
 
-const Title = styled.div`
-    font-weight: 600;
-    font-size: 12px;
-    line-height: 28px;
-    letter-spacing: 0.02em;
-    text-transform: uppercase;
-    padding: 0px 12px;
-    color: rgba(var(--center-channel-color-rgb), 0.56);
-`;
-
-const Members = styled.div`
-    &.editing-transition-enter {
-        .member-role-chooser {
-            opacity: 0;
-            display: block;
-        }
-    }
-
-    &.editing-transition-enter-active {
-        .member-role-chooser {
-            opacity: 1;
-            transition: opacity 250ms;
-        }
-    }
-
-    &.editing-transition-enter-done {
-        .member-role-chooser {
-            opacity: 1;
-        }
-    }
-
-    &.editing-transition-exit {
-        .member-role-chooser {
-            opacity: 1;
-            display: block;
-        }
-    }
-
-    &.editing-transition-exit-active {
-        .member-role-chooser {
-            opacity: 0;
-            transition: opacity 250ms;
-            display: block;
-        }
-    }
-`;
-
 export interface Props {
-    className?: string;
     channel: Channel;
     members: ChannelMember[];
     editing: boolean;
-    title?: JSX.Element;
+    hasNextPage: boolean;
+    isNextPageLoading: boolean;
 
     actions: {
         openDirectMessage: (user: UserProfile) => void;
+        loadMore: () => void;
     };
 }
 
-const MemberList = ({className, channel, members, editing, title, actions}: Props) => {
-    return (
-        <div className={className} >
-            {members.length > 0 && (
-                <>
-                    {Boolean(title) && (<Title>{title}</Title>)}
+const MemberList = ({
+    hasNextPage,
+    isNextPageLoading,
+    channel,
+    members,
+    editing,
+    actions,
+}: Props) => {
+    // If there are more items to be loaded then add an extra row to hold a loading indicator.
+    const itemCount = hasNextPage ? members.length + 1 : members.length;
+
+    // Only load 1 page of items at a time.
+    // Pass an empty callback to InfiniteLoader in case it asks us to load more than once.
+    const loadMoreItems = isNextPageLoading ? () => {} : actions.loadMore;
```

Comment:
Should this be wrapped in a `useCallback` so that the `InfiniteLoader` only re-renders when this actually changes?

Reply:
I'm not sure it would be beneficial here, `action.loadMore` does not change nor take arguments, so the call is always the same.

Reply:
It's more that we're passing a new empty function as `loadMoreItems` any time `isNextPageLoading` is true. That means whenever MemberList renders, the InfiniteLoader will as well. Maybe that's not actually a concern though since the AutoSizer might do that anyway since I'm not sure how it actually works under the hood